schema-dex-amm-extended.graphql

```diff
@@ -611,62 +650,182 @@ type LiquidityPoolHourlySnapshot @entity {
+##############################
 
-"""
-An event is any user action that occurs in a protocol. Generally, they are Ethereum events
-emitted by a function in the smart contracts, stored in transaction receipts as event logs.
-However, some user actions of interest are function calls that don't emit events. For example,
-the deposit and withdraw functions in Yearn do not emit any events. In our subgraphs, we still
-store them as events, although they are not technically Ethereum events emitted by smart
-contracts.
-"""
-interface Event {
-  " { Event type }-{ Transaction hash }-{ Log index } "
+type Account @entity {
+  " { Account address } "
   id: ID!
 
-  " Transaction hash of the transaction that emitted this event "
+  " Number of positions this account has "
+  positionCount: Int!
+
+  " All positions that belong to this account "
+  positions: [Position!]! @derivedFrom(field: "account")
+
+  " All open positions this account has "
+  openPositions: [Position!]!
+
+  " Number of open positions this account has "
+  openPositionCount: Int!
+
+  " All closed positions this account has "
+  closedPositions: [Position!]!
+
+  " Number of closed positions this account has "
+  closedPositionCount: Int!
+
+  " Number of deposits this account made "
+  depositCount: Int!
+
+  " All deposit events of this account "
+  deposits: [Deposit!]! @derivedFrom(field: "account")
+
+  " Number of withdrawals this account made "
+  withdrawCount: Int!
+
+  " All withdraw events of this account "
+  withdraws: [Withdraw!]! @derivedFrom(field: "account")
+
+  " Number of times this account has traded/swapped"
+  swapCount: Int!
+
+  " All swap events of this account "
+  swaps: [Swap!]! @derivedFrom(field: "account")
+}
+
+type Position @entity {
+  " { Account address }-{ Market address } "
+  id: ID!
+
+  " Account that owns this position "
+  account: Account!
+
+  " The liquidity pool in which this position was opened "
+  pool: LiquidityPool!
+
+  " The hash of the transaction that opened this position "
+  hashOpened: String!
+
+  " The hash of the transaction that closed this position "
+  hashClosed: String
+
+  " Block number of when the position was opened "
+  blockNumberOpened: BigInt!
+
+  " Timestamp when the position was opened "
+  timestampOpened: BigInt!
+
+  " Block number of when the position was closed (0 if still open) "
+  blockNumberClosed: BigInt
+
+  " Timestamp when the position was closed (0 if still open) "
+  timestampClosed: BigInt
+
+  " Input token balances in this position, in native amounts "
+  inputTokenBalances: [BigInt!]!
+
+  " Output token balance in this position, in native amounts "
+  outputTokenBalance: BigInt
+
+  " Total reward token accumulated under this position, in native amounts "
+  cumulativeRewardTokenAmounts: [BigInt!]
+
+  " Number of deposits related to this position "
+  depositCount: Int!
+
+  " All deposit events of this position "
+  deposits: [Deposit!]! @derivedFrom(field: "position")
```

Comment:
There is no field `position` in `Withdraw` and `Deposit`. Needs to be updated.

Reply:
updated